src/librustc/ty/instance.rs

```diff
@@ -25,6 +26,9 @@ pub enum InstanceDef<'tcx> {
     /// `<T as Trait>::method` where `method` receives unsizeable `self: Self`.
     VtableShim(DefId),
 
+    /// `fn()` pointer where the function is annotated with `#[track_caller]`.
```

Comment:
I think the comment should be more general, saying something along the lines of "where the function itself cannot be turned into a pointer", with `#[track_caller]` given as an example.

Reply:
ack